projects/ngx-avatar/src/lib/avatar.component.ts

```diff
@@ -153,18 +151,34 @@ export class AvatarComponent implements OnChanges, OnDestroy {
    *
    * memberOf AvatarComponent
    */
-  public fetchAvatarSource(event?: any): void {
-    const avatarSource = this.sources[this.currentSource];
-    if (!avatarSource) {
+  public fetchAvatarSource(): void {
+    const previousSource = this.sources[this.currentIndex];
+    if (previousSource) {
+      this.avatarService.markSourceAsFailed(previousSource);
+    }
+
+    const source = this.findNextSource();
+    if (!source) {
       return;
     }
-    if (this.avatarService.isTextAvatar(avatarSource.sourceType)) {
-      this.buildTextAvatar(avatarSource);
+
+    if (this.avatarService.isTextAvatar(source.sourceType)) {
+      this.buildTextAvatar(source);
       this.avatarSrc = null;
     } else {
-      this.buildImageAvatar(avatarSource);
+      this.buildImageAvatar(source);
+    }
+  }
+
+  private findNextSource(): Source | null {
+    while (++this.currentIndex < this.sources.length) {
```

Comment:
Wouldn't something like...

```suggestion
    return this.sources.some(source => !this.avatarService.sourceHasFailedBefore(source));
```

... be more effective?

We still can increment `currentIndex` if needed:

```typescript
    return this.sources.some(source => {
      this.currentIndex++;
      return !this.avatarService.sourceHasFailedBefore(source)
    });
```

Reply:
I don't think this would work, because you would start over from the beginning of the sources every time.

Given an array of sources similar to:
```
[failed, failed, failed-current, next, next]
```

The current version would typically do one iteration from `failed-current` to find the first `next`. But the suggested version would re-iterate over the same first `failed` avatars again and again.

Reply:
I see, I didn't think of that as it always starts in the `this.currentIndex`.